### PR TITLE
Implement detection for  cve_2024_31461

### PR DESCRIPTION
**Pull request by PiranhaSa**

### CODE REVIEW

File: agent/exploits/cve_2024_31461.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+from agent import exploits_registry
+
+VULNERABILITY_TITLE = "SSRF flaw in Plane Project Management Software"
+VULNERABILITY_REFERENCE = "CVE-2024-31461"
+VULNERABILITY_DESCRIPTION = (
+    "Plane, an open-source project management tool, has a Server-Side Request Forgery (SSRF) vulnerability in versions prior to 0.17-dev. This "
+    "issue may allow an attacker to send arbitrary requests from the server hosting the application, potentially leading to unauthorized access to internal systems. "
+)
+
+DEFAULT_TIMEOUT = 30
```
Comment (elyousfi5) [inline agent/exploits/cve_2024_31461.py:19]:
```suggestion
DEFAULT_TIMEOUT = datetime.timedelta(seconds=30)
```

File: agent/exploits/cve_2024_31461.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    CVE-2024_31461: SSRF flaw in Plane Project Management Software.
+    """
+
+    def accept(self, target: definitions.Target) -> bool:
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        try:
+            resp = session.get(target.url, timeout=DEFAULT_TIMEOUT)
```
Comment (elyousfi5) [inline agent/exploits/cve_2024_31461.py:35]:
```suggestion
            resp = session.get(target.url, timeout=DEFAULT_TIMEOUT.total_seconds())
```

File: agent/exploits/cve_2024_31461.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        session = requests.Session()
+        session.max_redirects = MAX_REDIRECTS
+        session.verify = False
+
+        vulnerabilities: list[definitions.Vulnerability] = []
+
+        try:
+            resp = session.get(target.url, timeout=DEFAULT_TIMEOUT)
```
Comment (elyousfi5) [inline agent/exploits/cve_2024_31461.py:52]:
```suggestion
            resp = session.get(target.url, timeout=DEFAULT_TIMEOUT.total_seconds())
```

File: tests/exploits/cve_2024_31461_test.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+"""
+
+
+def testCVE202431461_whenVulnerable_reportFinding(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """CVE-2024-31461 unit test: case when target is vulnerable."""
+    requests_mock.get(
+        "http://localhost:80/",
+        text=INSECURE_HTML,
```
Comment (elyousfi5) [inline tests/exploits/cve_2024_31461_test.py:86]: put INSECURE_HTML in a fixture !!

File: tests/exploits/cve_2024_31461_test.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+
+def testCVE202431461_whenSafe_reportNothing(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """CVE-2024-31461 unit test: case when target is safe."""
+    exploit_instance = cve_2024_31461.CVE202431461Exploit()
+    requests_mock.get(
+        "http://localhost:80/",
+        text=SECURE_HTML,
```
Comment (elyousfi5) [inline tests/exploits/cve_2024_31461_test.py:111]: same

File: agent/exploits/cve_2024_31461.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            if resp.status_code == 200 and "EmojiPickerReact" not in resp.text:
```
Comment (3asm) [inline agent/exploits/cve_2024_31461.py:53]: Can you please add a comment justifying why are you looking for this string.

**Review by 3asm — APPROVED** [review]

File: agent/exploits/cve_2024_31461.py (hunk 6 above)
Comment (PiranhaSa) [reply agent/exploits/cve_2024_31461.py:56]: done

**Review by rabsondev — APPROVED** [review]

**Review by ErebusZ — APPROVED** [review]

**Pull request merged**